### PR TITLE
[5.2] Allow for @has in blade

### PR DESCRIPTION
**Pull request by alexbowers**
Currently, it is difficult to determine whether something can be @yield'd

Code usage is the same as @if.

Example usage:

image

### CODE REVIEW

**Comment by alexbowers** [comment]
Okay. styleci makes things annoying. If necessary I can squash all commits down.

**Comment by alexbowers** [comment]
I also can't find where blade gets tested.

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 1]
```diff
@@ -662,6 +662,17 @@ protected function compileElseif($expression)
...
 
     /**
+     * Compile the has statements into valid PHP.
+     *
+     * @param  string  $expression
+     * @return string
+     */
+    protected function compileHas($expression)
+    {
+        return "<?php if(!empty(trim(\$__env->yieldContent{$expression}))): ?>";
```
Comment (mark86092) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: You may use $__env->hasSection to determine whether a section exists.

Actually, yieldContent does check the section is set

Comment (alexbowers) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: So should this be changed? To me this seems correct. I want to see whether there is any content to be yielded.

**Comment by mark86092** [comment]
IMHO, you should define the `@has`, is it "has section" or "is yielding content nothing" ?

For example, if a section foo
```
@section("foo", "0 ");
```
, the `@has("foo")` results in false. (`empty(trim("0 ") --> true`)

**Comment by alexbowers** [comment]
Ah yes. Perhaps the fix is to check length > 0, rather than empty?

**Comment by alexbowers** [comment]
And should it perhaps be called hasYield?

**Comment by Cyrille37** [comment]
Intuitively, I think `@has` is to test if a **view variable exists** ...
I really prefer `@hasYield`

**Comment by themsaid** [comment]
I have a custom blade directive in one of my projects that works like this:

```
@display('title', 'Site Name', '% - Site Name')
```

It yields the section content as it is if exists, if not then the second argument will be displayed. If the third argument has a value it'll yield the section content based on the format.

Maybe such directive can be used to provide the same functionality instead of a condition, otherwise I believe `@hasSection` should end with `@endhassection` just like `@endcan`. Of maybe make it `@ifHasSection` so that it may end with `@endif`

**Comment by mark86092** [comment]
I would like `@hassection` with `@endsection` or `@hasyield` with `@endhasyield`. 
Although the word is long, it is more clear

**Comment by alexbowers** [comment]
Before I make any changes on this then, I'd like to get @taylorotwell input on this. Which direction do you think this should go (if any?).

**Comment by taylorotwell** [comment]
I believe you can also already do `@yield('title', 'Site Name');`

Does that solve your problem?

**Comment by mark86092** [comment]
@taylorotwell, the problem is something must show if title section exist, not your example something must show if title doesn't exist

**Comment by alexbowers** [comment]
Yeah, the use case here for me is a conditional one.

**Comment by taylorotwell** [comment]
Renamed to `@hasSection`

**Comment by vedmant** [comment]
Great! Today I actually was searching for something like this, in my case I needed to put default HTML content for section, using second argument for `@yield` doesn't go easily with more complex HTML or if it has blade directives.

**Comment by alexbowers** [comment]
@taylorotwell Do you want a PR with `@endhassection` in it too? Or has that been added by yourself already / you're happy with `@endif`

**Pull request merged**